### DISCUSSION TITLE
Using Daily files, Rotating files with async at the same time

### DISCUSSION THREAD

**Discussion by penguin-madagascar** [gabime/spdlog#2776] [Q&A]
Hi, I am new to spdlog. I saw that we can use spdlog with _daily_file_sink_ or _rotating_file_sink_ to customize the output. 
I am wondering if we can us the _daily_file_sink_  with _rotating_file_sink_  in mutithread(async) condition, which means the logger will create a file everyday, with a size limit, when it reach the size limit, it will create a new file, like a rotating fille logger.

Is there any way to realize that?

Thanks in advance!

**Comment by tt4g** [comment]
See #2702, #1393

**Comment by penguin-madagascar** [comment]
my solution for now is to manully assign the filename of the log:
```
std::string getCurrentDate() {
    std::time_t now = std::time(nullptr);
    std::tm* timeInfo = std::localtime(&now);

    std::ostringstream oss;
    oss << std::put_time(timeInfo, "%Y_%m_%d");

    return oss.str();
}
```
```
//if date is different from recorded date:
void async_logger_init(std::string log_filename, int max_filesize, int max_filenum, int pool_size) {
        spdlog::drop_all();

        spdlog::init_thread_pool(pool_size, std::thread::hardware_concurrency()-1);
        auto async_sink = std::make_shared<spdlog::sinks::rotating_file_sink_mt>("logs/" + getCurrentDate() + ".txt", 1024 * 1024 * max_filesize, max_filenum);
        auto async_logger = std::make_shared<spdlog::async_logger>("async_logger", async_sink, spdlog::thread_pool(), spdlog::async_overflow_policy::block);
        spdlog::register_logger(async_logger);
}
```

Please note that this may cause various issues or slow down the entire process, a better solution should be create your own sink: https://github.com/gabime/spdlog/wiki/4.-Sinks#implementing-your-own-sink
